tests/test_cli_compile.py

```diff
@@ -2030,6 +2030,31 @@ def test_preserve_compiled_prerelease_version(pip_conf, runner):
     assert "small-fake-a==0.3b1" in out.stderr.splitlines()
 
 
+def test_ignore_compiled_unavailable_version(pip_conf, runner, current_resolver):
+    if current_resolver == "legacy":
+        pytest.xfail(
+            "We know this is broken in the legacy resolver, but no fix is planned."
+        )
```

Comment:
```suggestion
backtracking_resolver_only = pytest.mark.parametrize(
    "current_resolver",
    ("backtracking",),
    indirect=("current_resolver",),
)

@backtracking_resolver_only
def test_ignore_compiled_unavailable_version(pip_conf, runner, current_resolver):
```

Can be done same way as `legacy_resolver_only`.